### DISCUSSION TITLE
Error: EMFILE: too many open files, open '/var/task/node_modules/iconoir-react/dist/esm/RemoveSelection.mjs'

### DISCUSSION THREAD

**Discussion by mattp278** [vercel/community#2865] [Help] [answered]
### Summary

You can visit this link.

https://nextjs-food-take-away-plzyctumz-mattpowell2784vercel-gmailcom.vercel.app/

https://nextjs-food-take-away-plzyctumz-mattpowell2784vercel-gmailcom.vercel.app/pages/auth/signin

Error: EMFILE: too many open files, open '/var/task/node_modules/iconoir-react/dist/esm/RemoveSelection.mjs'

### Steps to Reproduce

Hello, all members.

I am using Next js and next-auth.

It is working fine locally, but when i deploy it on vercel, i am getting errors when I visit sign in page.

```
[Error: EMFILE: too many open files, open '/var/task/node_modules/iconoir-react/dist/esm/RemoveSelection.mjs'] {
  errno: -24,
  code: 'EMFILE',
  syscall: 'open',
  path: '/var/task/node_modules/iconoir-react/dist/esm/RemoveSelection.mjs',
  page: '/pages/auth/signin'
}
RequestId: 9dc0b008-77f1-4855-bf2a-53bcd1d6c832 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

Any helpful suggestion will be great, thanks.

<img width="2992" alt="Screenshot 2023-06-14 at 18 06 15" src="https://github.com/vercel/community/assets/136371841/1620f7e9-cc99-46a3-af92-ebed6d38613b">

<img width="2992" alt="Screenshot 2023-06-14 at 18 06 43" src="https://github.com/vercel/community/assets/136371841/0057d991-95a6-4867-b78a-39a539ec3360">

**Comment by mbenchekroun** [comment]
Hello,  I saw your post on freelancer:
Increase the file limit: Vercel's serverless environment has a limit on the number of open files. You can try increasing this limit by setting the VERCEL_MAX_ASSETS environment variable to a higher value. For example, you can set it to 1000 by adding the following to your Vercel project's configuration file (vercel.json or now.json): this issue is due to you loading lot of files. (freelancer username mbenchekroun)

**Comment by amyegan** [comment, marked as answer]
The EMFILE error means that the build work exceeds the Vercel builder limits. Since the error seems to be related to `iconoir-react`, you might be able to find a different way of importing or try building without the icons. I'm not very familiar with how `iconoir-react` works, so it might be worth also asking the Iconoir for advice if you have not already. 

Alternatively, you can use [Vercel CLI](https://vercel.com/docs/cli#) locally to build and upload the output by following this guide: https://vercel.com/docs/cli/deploying-from-cli#deploying-from-local-build-prebuilt
